Assets/Scripts/R&D Tree/ResearchNode.cs

```diff
@@ -0,0 +1,134 @@
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+public class ResearchNode : MonoBehaviour
+{
+    [SerializeField] private Button _button;
+    [SerializeField] private Slider _researchBar;
+
+    private float _timer;
+
+    public UnityEvent ResearchDoneEvent = new UnityEvent();
+
+    private ResearchNodeSetting _nodeSetting;
+
+    public SkillState CurrentSkillState;
+
+    private void OnEnable()
+    {
+        SetUIForStates();
+    }
+
+    private void Start()
+    {
+        _nodeSetting = GetComponent<ResearchNodeSetting>();
+
+        _researchBar.maxValue = _nodeSetting.ResearchTime;
+        AddListenersToEvent();
+    }
+
+    /// <summary>
+    /// This function is called when the researchnode started to be researched
+    /// </summary>
+    public void TimerForInDevelopment()
+    {
+        if (CurrentSkillState != SkillState.inDevelopment) return;
+
+        ///TODO: Check if there is enough money
+
+        _timer += Time.fixedDeltaTime;
+        _researchBar.value = _timer;
+        if (_timer >= _nodeSetting.ResearchTime)
+        {
+            ResearchDoneEvent.Invoke();
+        }
+    }
+
+    /// <summary>
+    /// This function is called when the research is done
+    /// </summary>
+    private void ResearchDone()
```

Comment:
Remove circle dependency by making I.E. a callback. Callback could also be passed through the initalizer or constructor